### PR TITLE
All merging of RenderableNodes, such that we can merge multiple topologies.

### LINKED ISSUE #171: Add Container topology
After #123 is merged, add a new Containers topology, so we can populate it with Containers that don't have any active connections, and also to have a target to attach arbitrary container metadata.

### PR DESCRIPTION
**Pull request by tomwilkie**
With this change, the containers topology shows containers not actively communicating.

Fixes #171

### CODE REVIEW

**Comment by tomwilkie** [comment]
@peterbourgon Early WIP, seems to work.

File: report/mapping.go [hunk 1]
```diff
@@ -77,11 +82,28 @@ func ProcessName(_ string, m NodeMetadata) (MappedNode, bool) {
 	}, show
 }
 
-// ProcessContainer maps topology nodes to the containers they run in. We
+// ProcessEndpoint2Container maps endpoint topology nodes to the containers they run in. We
 // consider container and image IDs to be globally unique, and so don't scope
 // them further by e.g. host. If no container metadata is found, nodes are
 // grouped into the Uncontained node.
```
Comment (peterbourgon) [inline report/mapping.go]: Please re-wrap

File: report/mapping.go [hunk 2]
```diff
@@ -77,11 +82,28 @@ func ProcessName(_ string, m NodeMetadata) (MappedNode, bool) {
...
+	return MappedNode{
+		ID:    id,
+		Major: major,
+		Minor: minor,
+		Rank:  rank,
+	}, true
+}
+
+// ProcessContainer2Container maps container topology node to container mapped nodes.
+func ProcessContainer2Container(_ string, m NodeMetadata) (MappedNode, bool) {
```
Comment (peterbourgon) [inline report/mapping.go]: - "Process" is a vestige of the old topology naming scheme, let's drop it
- This is effectively an identity mapping, right? Can we call it ContainerIdentity?

File: report/mapping.go [hunk 3]
```diff
@@ -77,11 +82,28 @@ func ProcessName(_ string, m NodeMetadata) (MappedNode, bool) {
 	}, show
 }
 
-// ProcessContainer maps topology nodes to the containers they run in. We
+// ProcessEndpoint2Container maps endpoint topology nodes to the containers they run in. We
 // consider container and image IDs to be globally unique, and so don't scope
 // them further by e.g. host. If no container metadata is found, nodes are
 // grouped into the Uncontained node.
-func ProcessContainer(_ string, m NodeMetadata) (MappedNode, bool) {
+func ProcessEndpoint2Container(_ string, m NodeMetadata) (MappedNode, bool) {
```
Comment (peterbourgon) [inline report/mapping.go]: "Process" is a vestige of the old topology naming scheme, let's drop it.

File: app/api_topology.go [hunk 4]
```diff
@@ -30,10 +30,19 @@ type APIEdge struct {
...
 }
 
+func render(rpt report.Report, maps []topologyMapper) report.RenderableNodes {
+	result := report.RenderableNodes{}
+	for _, m := range maps {
+		rns := m.selector(rpt).RenderBy(m.mapper, m.pseudo)
+		result.Merge(rns)
+	}
+	return result
+}
```
Comment (peterbourgon) [inline app/api_topology.go]: Just so I've got it straight in my head — this is the meat of the change, right? This is the bit that flattens all of the metadatas into each other?

Comment (tomwilkie) [inline app/api_topology.go]: Yes the rest is plumbing.

Going forward, once we can chain these mapping functions, you would make this recursive.

**Comment by peterbourgon** [comment]
OK, assuming the stuff we chatted about is for a subsequent PR, and after those things are fixed, LGTM.

**Pull request merged**